submissions/xT0PERx/OOP exercise/index.js

```diff
@@ -0,0 +1,93 @@
+class Mammals {
+  constructor(props) {
+    this.name = props.name;
+    this.saying = props.saying;
+  }
+  say() {
+    return this.saying;
+  }
+  printInfo() {
+    print(
+      [
+        this.species,
+        this.name,
+        this.gender,
+        this.hands,
+        this.legs,
+        this.say(),
+      ].join('; ')
+    );
+  }
+}
+
+class Animal extends Mammals {
+  constructor({ name, gender, legs = 4, saying }) {
+    super({ name, saying });
+    this.species = 'animal';
+    this.gender = gender;
```

Comment:
All child classes have `species` and `gender` props. This is a clear sign that it should be initialized and handled by a super class.
Double check other classes and other props.

Reply:
@xT0PERx cannot identify changes addressing the issue above. Please fix or ask a question

Reply:
There is a question. As I thought when creating the superclass and classes:
1. I have a basic constructor - mammals. He creates a mammal - but he does not yet know whether it will be a human or an animal (I decided to divide mammals this way).
2. Then I create classes of people and animals from my superclass (here I add the species)
3. The following are already specified classes such as cat, dog, etc.
What's wrong with my logic? Why should I add species to the Mammals?
Perhaps during the day I myself will reach what you meant, it also happens. But for now, I would like to hear your thought. Thank you.